packages/client/src/BrubeckNode.ts

```diff
@@ -55,13 +56,29 @@ export default class BrubeckNode implements Context {
         this.destroySignal.assertNotDestroyed(this)
     }
 
+    private async getNormalizedNetworkOptions(): Promise<NetworkNodeOptions> {
+        if ((this.options.trackers as TrackerRegistrySmartContract).contractAddress) {
+            const trackerRegistry = await getTrackerRegistryFromContract({
+                contractAddress: (this.options.trackers as TrackerRegistrySmartContract).contractAddress,
+                jsonRpcProvider: this.ethereum.getMainnetProvider()
+            })
+            return {
+                ...this.options,
+                trackers: trackerRegistry.getAllTrackers()
+            }
+        }
+        return this.options as NetworkNodeOptions
+    }
+
     private async initNode() {
         this.assertNotDestroyed()
         if (this.cachedNode) { return this.cachedNode }
 
         const { options } = this
         let { id } = options
 
+        const networkOptions = await this.getNormalizedNetworkOptions()
```

Comment:
Could do this right before `createNetworkNode()` call to make the flow super readable. Then it is also possible remove lines 54 & 67 from `BrubeckNode.test.ts`.